lib/worker-pipeline.js

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const debug = console.log;
+const { Engine, Plan } = require('@nui/asset-compute-pipeline');
+const WorkerCallbackTransformer = require('./worker-transformer');
+
+const fs = require('fs-extra');
+const path = require('path');
+
+/**
+ * Represents a pipeline build around a worker callback.
+ * Mostly here for backwards compatibility.
+ */
+class AssetComputeWorkerPipeline {
+    constructor(renditionCallback, options={}){
+        this.options = options;
+
+        this.transformers = [];
+        if(this.options.transformerCatalog){
+            debug("Adding transformers from transformer catalog to the pipeline");
+            this.transformers = this.transformers.concat(options.transformerCatalog);
+        }
+
+        this.buildTransformersFromManifests(renditionCallback);
+        debug("Built worker transformer using rendition callback 'renditionCallback'");
+    }
+
+    /**
+     * Builds transformers from manifests defined in the options object
+     * @param {*} renditionCallback worker rendition callback to use in the built transformer
+     * @param {*} options options, must contain manifests (if not - will not create any transformer)
+     * @returns a list of transformers available for this pipeline instance
+     */
+    buildTransformersFromManifests(renditionCallback) {
+        if (!this.options.manifests) {
+            debug("No additional manifest in the pipeline options, no additional WorkerCallbackTransformer added to pipeline");
+            return this.transformers;
+        }
+
+        const manifests = this.options.manifests;
+        debug('Manifest array', manifests);
+    
+        const transformers = [];
+        manifests.forEach(manifest => {
+            const transformer = new WorkerCallbackTransformer(renditionCallback, manifest);
+
+            debug(`Created a transformer named ${transformer.name} with manifest:`);
+            debug(manifest);
+
+            transformers.push(transformer);
+        });
+
+        debug(`Adding ${transformers.length} WorkerCallbackTransformer to the pipeline`);
+        this.transformers = transformers.concat(this.options.transformerCatalog);
+    }
+
+    prepareEngine(params){
+        debug("Preparing pipeline engine... ");
+        const engine = new Engine(params);
+
+        this.transformers.forEach(transformer => {
+            console.log(`Registering transformer ${transformer.name} to the pipeline's engine`);
+            engine.registerTransformer(transformer);
+        });
+
+        return engine;
+    }
+
+    async compute(params){
+        const engine = this.prepareEngine(params);
```

Comment:
If its one rendition per engine, we would need an engine for each rendition.

Reply:
note: ticket to implement support for multiple renditions in pipeline mode is in the next sprint. it does not need to be handled in this ticket

see ticket here: https://jira.corp.adobe.com/browse/NUI-1284